presto-main/src/main/java/com/facebook/presto/operator/PagesSpatialIndexSupplier.java

```diff
@@ -137,7 +146,19 @@ private static STRtree buildRTree(LongArrayList addresses, List<List<Block>> cha
                 partition = toIntExact(INTEGER.getLong(partitionBlock, blockPosition));
             }
 
-            rtree.insert(getJtsEnvelope(ogcGeometry, radius), new GeometryWithPosition(ogcGeometry, partition, position));
+            GeometryWithPosition geometryWithPosition = new GeometryWithPosition(ogcGeometry, partition, position);
+            rtree.insert(getJtsEnvelope(ogcGeometry, radius), geometryWithPosition);
+
+            // 8-bytes for reference to the new element in itemBoundables array of STRtree
+            long itemBoundableMemorySizeInBytes = ITEM_BOUNDABLE_INSTANCE_SIZE + ENVELOPE_INSTANCE_SIZE + geometryWithPosition.getEstimatedMemorySizeInBytes() + 8;
+
+            estimatedRTreeMemorySizeInBytes += itemBoundableMemorySizeInBytes;
+
+            // update estimated memory size when change is more than 100 MB
+            if (estimatedRTreeMemorySizeInBytes - lastUpdatedRTreeMemorySizeInBytes >= 100_000_000) {
```

Comment:
Use a constant for 100MB

`private static final int MEMORY_USAGE_UPDATE_INCREMENT_BYTES = 100 * 1024 * 1024;   // 100 MB`